cordapp-cpk/src/main/kotlin/net/corda/plugins/cpb/CpbPlugin.kt

```diff
@@ -0,0 +1,52 @@
+package net.corda.plugins.cpb
+
+import net.corda.plugins.cpk.CORDAPP_CONFIGURATION_NAME
+import net.corda.plugins.cpk.CORDA_CPK_CONFIGURATION_NAME
+import net.corda.plugins.cpk.CordappPlugin
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.ExternalDependency
+import org.gradle.api.artifacts.ProjectDependency
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.api.tasks.bundling.ZipEntryCompression
+
+class CpbPlugin : Plugin<Project> {
+
+    companion object{
+        private const val CPB_TASK_NAME = "cpb"
+        private const val CPB_CONFIGURATION_NAME = CPB_TASK_NAME
+        private const val CPB_ARTIFACT_CLASSIFIER = "package"
+        private const val CPB_FILE_EXTENSION = CPB_TASK_NAME
+    }
+
+    override fun apply(project: Project) {
+        project.plugins.findPlugin(CordappPlugin::class.java) ?:
+            throw GradleException("The 'net.corda.plugins.cpb' plugin needs the 'net.corda.plugins.cordapp-cpk' plugin to be applied first")
```

Comment:
Just use
```kotlin
project.pluginManager.apply('net.corda.plugins.cordapp-cpk')
```
This will either apply the `cordapp-cpk` plugin, or do nothing if it has already been applied.